File: backend/src/main/java/com/cooknote/backend/domain/category/service/CategoryService.java
package com.cooknote.backend.domain.category.service;

import java.util.List;

import org.springframework.stereotype.Service;

import com.cooknote.backend.domain.category.dto.response.CategoryGetAllResponseDTO;
import com.cooknote.backend.domain.category.dto.response.CategoryReportReasonResponseDTO;
import com.cooknote.backend.domain.category.dto.response.CategoryResponseDTO;

@Service
public interface CategoryService {

	CategoryGetAllResponseDTO getCategoryAll();

	CategoryResponseDTO getCategory();

	List<CategoryReportReasonResponseDTO> getCategoryReportReason();

}

File: frontend/src/services/categoryService.js
import privateAPI from "@/api/privateAPI"
import publicAPI from "@/api/publicAPI"

// 레시피 카테고리 전부 가져오기
export const getCategoryAll = async () => {
  return await privateAPI.get(`/category/all`)
}

// 레시피 카테고리 가져오기
export const getCategory = async () => {
  return await publicAPI.get(`/category`)
}